src/app/api/configuration/route.ts

```diff
@@ -0,0 +1,50 @@
+import { NextRequest, NextResponse } from "next/server";
+import { options } from "../auth/[...nextauth]/options";
+import {
+  getTeamWeightConfigs,
+  saveTeamWeightConfig,
+} from "./configuartionSerice";
+import { getServerSession } from "next-auth";
+import {
+  WeightConfigDto,
+  WeightConfig,
+  defaultConfigResponse,
+  weightValueMap,
+} from "./weightConfig.dto";
+
+export async function GET(req: NextRequest) {
+  const session = await getServerSession(options);
+  const teamId = req.nextUrl.searchParams.get("team_id");
+  if (session?.userId && teamId) {
+    const weightConfigDto = await getTeamWeightConfigs(teamId);
+    return NextResponse.json({
+      success: true,
+      weights: mapToDto(weightConfigDto),
+    });
+  }
+  return NextResponse.json({ success: false, weights: [] });
+}
+
+export async function POST(req: NextRequest) {
+  const session = await getServerSession(options);
+  const teamId = req.nextUrl.searchParams.get("team_id");
+  if (session?.userId && teamId) {
+    const weightConfig = (await req.json()) as WeightConfig;
```

Comment:
Should we fetch the weightConfig before the if condition so we can check it as well as conditional? Query parameters should always be checked IMHO

Reply:
No need,  I insert default values if its not there